### PR TITLE
boards: intel_adsp: unify and update docs

### PR DESCRIPTION
**Pull request by laurenmurphyx64**
Unifies docs into one updated guide with stubs for each platform.

### CODE REVIEW

**Comment by nashif** [comment]
image

Does not look right.
also, both chromebook and general documentation should not appear here, we just want the platforms.

File: boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.. _intel_adsp_ace15:
+
+Intel ADSP ACE 15
```
Comment (nashif) [inline boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst:3]: 1.5

File: boards/xtensa/intel_adsp_cavs15/doc/index.rst [hunk 2]
```diff
@@ -1,156 +1,34 @@
-.. _Up_Squared_Audio_DSP:
+.. _intel_adsp_cavs15:
 
-Up Squared Audio DSP
-####################
+Intel ADSP CAVS 15
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs15/doc/index.rst:3]: 1.5

File: boards/xtensa/intel_adsp_cavs18/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.. _intel_adsp_cavs18:
+
+Intel ADSP CAVS 18
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs18/doc/index.rst:3]: 1.8

File: boards/xtensa/intel_adsp_cavs25/doc/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
+.. _intel_adsp_cavs25:
+
+Intel ADSP CAVS 25
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/doc/index.rst:3]: 2.5

File: boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+*****************
+
+Overview
+********
```
Comment (nashif) [inline boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst:7]: Wrong level

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by laurenmurphyx64** [comment]
> Does not look right. also, both chromebook and general documentation should not appear here, we just want the platforms.

I've hidden the pages and fixed the leveling. It isn't showing up correctly in the build preview (it's doing something weird with the caches), but when I download the HTML and open it it looks correct:

image

So I orphaned the generic ADSP doc and the Chromebook doc. They don't appear in any toctree since I wasn't sure where to put them. If you have a place in mind, let me know. Right now, the ADSP doc can be reached only through links from each of the board pages. The Chromebook page being out of date isn't reachable at all, but I can add a link to it in the CAVS 2.5 doc if you want.

image

File: boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+Intel ADSP ACE 1.5
+##################
+
+Overview
+********
+
+This board configuration is used to run Zephyr for ADSP applications on Intel
+Meteor Lake microprocessors. It is most commonly used to run version 1.5 of
+Intel's ACE firmware. Refer to :ref:`intel_adsp_generic` for more details.
```
Comment (kv2019i) [inline boards/xtensa/intel_adsp_ace15_mtpm/doc/index.rst:11]: s/firmware/hardware/ Version 1.5 is version of the hardware. Maybe reword as: "This board configuration is used to run Zephyr on the Intel ACE 1.5 Audio DSP. This configuration is present for example on Intel Meteor Lake microprocessors."

File: boards/xtensa/intel_adsp_cavs15/doc/index.rst [hunk 7]
```diff
@@ -1,156 +1,34 @@
...
-Signing tool
-------------
+Overview
+********
 
-rimage is Audio DSP firmware image creation and signing tool. The tool is used
-by `Sound Open Firmware`_ to generate binary firmware signed images.
+This board configuration is used to run Zephyr for ADSP applications on Intel
+`Apollo Lake`_ microprocessors. It is most commonly used to run version 1.5 of
+Intel's CAVS firmware. Refer to :ref:`intel_adsp_generic` for more details.
```
Comment (kv2019i) [inline boards/xtensa/intel_adsp_cavs15/doc/index.rst:11]: s/firmware/hardware/ here as well. Same wording could be used as above.

File: boards/xtensa/intel_adsp_cavs18/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
+.. _intel_adsp_cavs18:
+
+Intel ADSP CAVS 1.8
+###################
+
+Overview
+********
+
+This board configuration is used to run Zephyr for ADSP applications on Intel
+`Whiskey Lake`_ microprocessors. It is most commonly used to run version 1.8 of
```
Comment (kv2019i) [inline boards/xtensa/intel_adsp_cavs18/doc/index.rst:10]: cAVS1.8 ADSP is used in multiple Intel products. Whiskey Lake is one, but also present on Comet Lake and Coffee Lake.

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+| CAVS 1.5 | Apollo Lake  |
++----------+--------------+
+| CAVS 1.8 | Whiskey Lake |
++----------+--------------+
+| CAVS 2.5 | Tiger Lake   |
++----------+--------------+
+| ACE 1.5  | Meteor Lake  |
++----------+--------------+
+
+The boards in the Zephyr tree are named after the version of the firmware that
```
Comment (kv2019i) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:27]: So here too, it's not the firmware version but version of the audio DSP hardware. Otherwise table is correct.

**Review by kv2019i — COMMENTED** [review]
Oh this is excellent, thanks @laurenmurphyx64 for doing this! 

I have a few notes on platform naming (which arguably is quite confusing). So ACE1.5 and cAVS2.5 are versions of the audio DSP hardware and these are then integrated into products like Tiger Lake. Firmware version is really the Zephyr version in this context.

**Comment by laurenmurphyx64** [comment]
> Oh this is excellent, thanks @laurenmurphyx64 for doing this!
> 
> I have a few notes on platform naming (which arguably is quite confusing). So ACE1.5 and cAVS2.5 are versions of the audio DSP hardware and these are then integrated into products like Tiger Lake. Firmware version is really the Zephyr version in this context.

Ahh, that makes sense... I feel dumb now, haha. Thanks for your review and especially the clarification, and let me know if I've correctly captured the meaning in my latest update to the docs.

**Review by nashif — APPROVED** [review]

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS vx.x.x-xxx-gxxxxxxxxxxxx *****
+   Hello World! intel_adsp_cavs25
+
+Flashing for ACE
+----------------
+
+Flashing is not yet supported for ACE boards, as ACE boards are not yet
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:237]: `ACE boards` is not quite right, there is nothing of that sort. `platfroms with the ACE based Audio DSP` is more appropriate.

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+   sudo ./cavstool.py zephyr.ri
+
+You should see the following at the end of the log if you are successful:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS vx.x.x-xxx-gxxxxxxxxxxxx *****
+   Hello World! intel_adsp_cavs25
+
+Flashing for ACE
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:234]: Flashing for ACE based Audio DSP.

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+     runner_params:
+     - --remote-host=remotehostname:reqport
+
+Any of the arguments you would pass to Twister or West, you can pass with the
+hardware map. As mentioned previously, you can see the Intel ADSP runner
+arguments by passing the ``--context`` flag while flashing with West.
+
+Refer to :ref:`twister_script` for more information on hardware maps.
+
+Local Flashing for West and Twister on CAVS
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:217]: ... on CAVS based Audio DSP.

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+``west flash`` if the flash target exists for your board. To see a list
+of all arguments to the Intel ADSP runner, run the following after you have
+built the binary. There are multiple arguments related to signing, including a
+key argument.
+
+.. code-block:: console
+
+   west flash --context
+
+Remote Flashing for West and Twister on CAVS
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:137]: CAVS based Audio DSP

File: boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst [hunk 14]
```diff
@@ -0,0 +1,338 @@
...
+host running on the board's main CPU. It is possible (and recommended) for users
+to build the binary locally on their development machine and flash remotely,
+but the board itself must be capable of running the Python script that receives
+the binary sent over the network by West and flashes it. You should install a
+version of Linux that supports or comes with the current version of Python that
+Zephyr requires. Consider using Ubuntu 20.04, which comes with Python 3.8
+installed.
+
+Note that if you plan to use SOF on your board, you will need to build and
+install the modified Linux SOF kernel instead. It is recommended you follow the
```
Comment (nashif) [inline boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst:53]: ... instead of the default kernel.

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by teburd — DISMISSED** [review]
This is fantastic! I think @nashif has hit on most of what I noted already so I won't repeat too much. I do think its worth adding my two cents on the xtensa toolchain blocks on each page. It does imply this is the only way to build and that's just not true, so having a gcc build section would be quite nice. 

The docs read really well and covers everything in one nice spot. This is the doc I always wanted but never had on hand.

Fantastic work.

File: boards/xtensa/intel_adsp_cavs18/doc/index.rst [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+********
+
+This board configuration is used to run Zephyr on the Intel CAVS 1.8 Audio DSP.
+This configuration is present, for example, on Intel `Whiskey Lake`_ microprocessors.
+Refer to :ref:`intel_adsp_generic` for more details on Intel ADSP ACE and CAVS.
+
+System requirements
+*******************
+
+Xtensa Toolchain
```
[the remark this answers is not in the document]
Comment (laurenmurphyx64) [reply boards/xtensa/intel_adsp_cavs18/doc/index.rst:16]: Good catch, oops

**Comment by laurenmurphyx64** [comment]
@nashif I think this is ready to merge, made the requested edits - check out the preview  and let me know what you think https://builds.zephyrproject.io/zephyr/pr/56210/docs/boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.html#intel-adsp-generic

**Review by nashif — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Thanks @laurenmurphyx64 !

**Review by teburd — APPROVED** [review]

**Review by softwarecki — APPROVED** [review]

**Review by jxstelter — APPROVED** [review]

**Review by tmleman — APPROVED** [review]
